contracts/upgradeable_contracts/BasicHomeBridge.sol

```diff
@@ -0,0 +1,145 @@
+pragma solidity 0.4.24;
+
+import "../upgradeability/EternalStorage.sol";
+import "../libraries/SafeMath.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20Basic.sol";
+import "./Validatable.sol";
+import "../libraries/Message.sol";
+
+contract BasicHomeBridge is EternalStorage, Validatable {
+    using SafeMath for uint256;
+
+    event UserRequestForSignature(address recipient, uint256 value);
+    event AffirmationCompleted (address recipient, uint256 value, bytes32 transactionHash);
+    event SignedForUserRequest(address indexed signer, bytes32 messageHash);
+    event SignedForAffirmation(address indexed signer, bytes32 transactionHash);
+    event CollectedSignatures(address authorityResponsibleForRelay, bytes32 messageHash, uint256 NumberOfCollectedSignatures);
+
+    function executeAffirmation(address recipient, uint256 value, bytes32 transactionHash) external onlyValidator {
+        bytes32 hashMsg = keccak256(abi.encodePacked(recipient, value, transactionHash));
+        bytes32 hashSender = keccak256(abi.encodePacked(msg.sender, hashMsg));
+        // Duplicated deposits
```

Comment:
Is it correct to use deposit here still?

Reply:
nope. for Native to erc20 is deposit, for erc20_to_erc20 is withdraw

Reply:
👍